src/tools.rs

```diff
@@ -253,7 +253,7 @@ pub async fn get(
     version: Option<&str>,
     offline: bool,
     client_options: &HttpClientOptions,
-) -> Result<PathBuf> {
+) -> Result<(PathBuf, String)> {
```

Comment:
I am not a fan of having this as a mandatory return value. Maybe we can have a `get_details` function, which returns a struct (which we can extend later). And then have an implementation of `get` reusing that new function, dropping everything exception the path.